### PR TITLE
Story Promo List - Add border prop

### LINKED ISSUE #5594: Story Promo List - border bottom
**Describe the bug**
In sections with 13 promos, the last row with regular promos without images takes two columns within 600-1007px.  

Since we remove the `border-bottom` from the last `story-promo-list` item in the list, we end up having this situation where the penultimate item has a border, not looking great.

Screenshot 2020-02-18 at 16 08 46

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to a Front Page with a section which has 13 promos
2. Resize the window between `600` and `1007`px
3. Scroll down to where the promos without image are

**Expected behaviour**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Partly Resolves https://github.com/bbc/simorgh/issues/5594

**Overall change:** 
Add `border` prop to the Story Promo List component to be able to hide the `border-bottom` when needed.

After this gets merged, we will have to pass the `border` as false to the penultimate promo without an image to fix the issue happening on the Front Page.

**Code changes:**
- Add `border` prop to `StoryPromoLi` and set the default to `true`.
- Do not apply the `border` between `600px` and `1007px` when the `border` prop is `false`
- Add a new story using `Grid` to show an example of removing the border from the penultimate promo
- Add new snapshot passing border as `false`
- Update README

**Before**:
border-false

**After**:
border-true

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by OlgaLyubin — APPROVED** [review]
Looks great

File: packages/components/psammead-story-promo-list/README.md [hunk 1]
```diff
@@ -13,7 +13,8 @@ The `@bbc/psammead-story-promo-list` package is a set of two components, `StoryP
 <!-- prettier-ignore -->
 | Argument | Type | Required | Default | Example        |
 | -------- | ---- | -------- | ------- | -------------- |
-| children | node | Yes      | N/A     | `<StoryPromoLi><StoryPromo image={Image} info={Info} /></StoryPromoLi>` |
+| children | node | yes      | N/A     | `<StoryPromoLi><StoryPromo image={Image} info={Info} /></StoryPromoLi>` |
+| border   | bool | no       | true    | `false` |
```
Comment (OlgaLyubin) [inline packages/components/psammead-story-promo-list/README.md]:
```suggestion
| border   | bool | no       | `true`    | `false` |
```

File: packages/components/psammead-story-promo-list/src/index.jsx [hunk 2]
```diff
@@ -15,9 +15,19 @@ import {
 export const StoryPromoLi = styled.li.attrs({
   role: 'listitem',
 })`
+
   @media (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
-    border-bottom: 0.0625rem solid ${C_LUNAR};
+      border-bottom: 0.0625rem solid ${C_LUNAR};
```
Comment (Bopchy) [inline packages/components/psammead-story-promo-list/src/index.jsx]:
```suggestion
    border-bottom: 0.0625rem solid ${C_LUNAR};
```

**Review by Bopchy — APPROVED** [review]
Just left a suggestion on the extra spacing. Otherwise LGTM

**Comment by PriyaKR** [comment]
Looks good to me on storybook.

**Pull request merged**